demos/interactive_face_detection_demo_gapi/visualizer.cpp

```diff
@@ -0,0 +1,361 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "visualizer.hpp"
+
+// EmotionBarVisualizer
+EmotionBarVisualizer::EmotionBarVisualizer(std::vector<std::string> const& emotionNames, cv::Size size, cv::Size padding,
+                                     double opacity, double textScale, int textThickness):
+                                     emotionNames(emotionNames), size(size), padding(padding),
+                                     opacity(opacity), textScale(textScale), textThickness(textThickness),
+                                     internalPadding(0) {
+    auto itMax = std::max_element(emotionNames.begin(), emotionNames.end(), [] (std::string const& lhs, std::string const& rhs) {
+        return lhs.length() < rhs.length();
+    });
+
+    textSize = cv::getTextSize(*itMax, cv::FONT_HERSHEY_COMPLEX_SMALL, textScale, textThickness, &textBaseline);
+    ystep = (emotionNames.size() < 2) ? 0 : (size.height - 2 * padding.height - textSize.height) / (emotionNames.size() - 1);
+}
+
+cv::Size EmotionBarVisualizer::getSize() {
+    return size;
+}
+
+void EmotionBarVisualizer::draw(cv::Mat& img, std::map<std::string, float> emotions, cv::Point org, cv::Scalar fgcolor, cv::Scalar bgcolor) {
+    cv::Mat tmp = img(cv::Rect(org.x, org.y, size.width, size.height));
+    cv::addWeighted(tmp, 1.f - opacity, bgcolor, opacity, 0, tmp);
+
+    auto drawEmotion = [&](int n, std::string text, float value) {
+        cv::Point torg(org.x + padding.width, org.y + n * ystep + textSize.height + padding.height);
+
+        int textWidth = textSize.width + 10;
+        cv::Rect r(torg.x + textWidth, torg.y - textSize.height, size.width - 2 * padding.width - textWidth, textSize.height + textBaseline / 2);
+
+        cv::putText(img, text, torg, cv::FONT_HERSHEY_COMPLEX_SMALL, textScale, fgcolor, textThickness);
+        cv::rectangle(img, r, fgcolor, 1);
+        r.width = static_cast<int>(r.width * value);
+        cv::rectangle(img, r, fgcolor, cv::FILLED);
+    };
+
+    for (size_t i = 0; i< emotionNames.size(); i++) {
+        drawEmotion(i, emotionNames[i], emotions[emotionNames[i]]);
```

Comment:
Do you really need to use the lambda-function here?

Reply:
Exactly
Left this implementation for more similar implementations